src/order/ideal.lean

```diff
@@ -87,7 +95,9 @@ instance : partial_order (ideal P) := partial_order.lift coe ext
 
 /-- A proper ideal is one that is not the whole set.
     Note that the whole set might not be an ideal. -/
-class proper (I : ideal P) : Prop := (nuniv : (I : set P) ≠ set.univ)
+class proper (I : ideal P) : Prop := (ne_univ : (I : set P) ≠ set.univ)
+
+lemma proper_iff {I : ideal P} : proper I ↔ (I : set P) ≠ set.univ := ⟨λ h, h.1, λ h, ⟨h⟩⟩
```

Comment:
The [mk_iff](https://leanprover-community.github.io/mathlib_docs/attributes.html#mk_iff) attribute can automatically generate `iff` lemmas from `Prop`-structures.
```suggestion
@[mk_iff] class proper (I : ideal P) : Prop := (ne_univ : (I : set P) ≠ set.univ)
```